src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -76,22 +76,25 @@ def get_artifact_path(self, platform, chunk, artifact):
     def get_chunks(self):
         return list(set([f.split('_')[1] for f in os.listdir('ccov-artifacts')]))
 
-    def get_coverage_artifacts(self, suite=None, chunk=None):
+    def get_coverage_artifacts(self, platform=None, suite=None, chunk=None):
         files = os.listdir('ccov-artifacts')
 
         if suite is not None and chunk is not None:
             raise Exception('suite and chunk can\'t both have a value')
 
+        # Filter artifacts according to platform, suite and chunk.
         filtered_files = []
         for fname in files:
-            # If suite and chunk are None, return all artifacts.
-            # Otherwise, only return the ones which have suite or chunk in their name.
-            if (
-                   (suite is None and chunk is None) or
-                   (suite is not None and ('%s' % suite) in fname) or
-                   (chunk is not None and ('%s_code-coverage' % chunk) in fname)
-               ):
-                filtered_files.append('ccov-artifacts/' + fname)
+            if platform is not None and not fname.startswith('%s_' % platform):
+                continue
+
+            if suite is not None and ('%s' % suite) not in fname:
```

Comment:
Why not `str(suite)` here ?

Reply:
Right, I can even use `suite` directly since it's either `None` or a string.